src/rpcblockchain.cpp

```diff
@@ -2782,13 +2791,28 @@ bool VerifyCPIDSignature(std::string sCPID, std::string sBlockHash, std::string
     return bValid;
 }
 
-std::string SignBlockWithCPID(std::string sCPID, std::string sBlockHash)
+bool SignBlockWithCPID(const std::string& sCPID, const std::string& sBlockHash, std::string& sSignature, std::string& sError)
```

Comment:
It looks cleaner when function does not pass error message in out parameter. Use something like:
`void SignBlockWithCPID(const std::string& sCPID, const std::string& sBlockHash, std::string& sSignature);`
And make it throw exception with the message.

Reply:
if u and ravon agree to throwing instread of a error message or bool i can work it to such. let me know your guyses consensus